### PR TITLE
Offline app

### PR DESCRIPTION
**Pull request by ashifkhn**
Date: 29th March 2024

Developer Name: Ashif

---

## Description
In our application, we've implemented offline behavior to handle scenarios where the user might not have an internet connection. Now, we've enhanced this behavior further. Previously, when the data was being loaded, we displayed a loader indefinitely until the data arrived. However, we've recognized that sometimes the data might not be available at all, and it's important to inform the user about this instead of showing an endless loader.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

https://github.com/Real-Dev-Squad/mobile-app/assets/54736284/19c05699-e5f9-4383-954b-eec2c1ba4bd7

</details>

Screenshot 2024-03-29 at 01 08 00

## Test Coverage
<details>
<summary>Screenshot 1</summary>

</details>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
   },
   "dependencies": {
     "@react-native-async-storage/async-storage": "^1.15.16",
+    "@react-native-community/netinfo": "^5.9.9",
```
Comment (iamitprakash) [inline package.json:20]: why are we using old version

File: assets/offline.png
Comment (iamitprakash) [inline assets/offline.png:1]: can we use better icon or get it approved @ankushdharkar  please

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
...
     "@react-native-community/slider": "^4.4.3",
```
Comment (iamitprakash) [inline package.json:21]: why are we using ^ this

File: package.json (hunk 1 above)
Comment (ashifkhn) [reply package.json:20]: The latest version of the package is not supported with the current RN version of our project.

File: assets/offline.png
Comment (ashifkhn) [reply assets/offline.png:1]: sure.

File: src/Index.tsx [hunk 3]
```diff
@@ -1,15 +1,43 @@
...
 const Index = () => {
   const { isLoading, loggedInUserData } = useContext(AuthContext);
+  const [isConnected, setIsConnected] = useState(false);
+
+  const retryConnection = async () => {
+    try {
+      const state = await NetInfo.fetch();
+
+      setIsConnected(state.isConnected);
+      console.log(state.isConnected);
```
Comment (prakashchoudhary07) [inline src/Index.tsx]: console.log?

File: src/constants/strings/strings.tsx [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export const strings = {
+  OFFLINE_TEXT: "Oops! Seems line you're offline",
```
Comment (prakashchoudhary07) [inline src/constants/strings/strings.tsx]: line?

File: __tests__/ConnectionScreen-test.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+describe('ConnectionScreen component', () => {
+  it('renders properly and triggers retry function on button press', () => {
+    const retryConnectMock = jest.fn();
+    render(<ConnectionScreen retryConnect={retryConnectMock} />);
+    expect(screen.getByText('Retry')).toBeTruthy();
+    fireEvent.press(screen.getByText('Retry'));
+    expect(retryConnectMock).toHaveBeenCalledTimes(1);
+  });
+});
```
Comment (prakashchoudhary07) [inline __tests__/ConnectionScreen-test.tsx:13]: Can we have tests for screen when internet is connected, and when internet is not connected, and when internet is back

File: src/constants/strings/strings.tsx (hunk 4 above)
Comment (ashifkhn) [reply src/constants/strings/strings.tsx]: fixed

File: src/Index.tsx (hunk 3 above)
Comment (ashifkhn) [reply src/Index.tsx]: fixed

File: __tests__/ConnectionScreen-test.tsx (hunk 5 above)
Comment (ashifkhn) [reply __tests__/ConnectionScreen-test.tsx:13]: Yes we can. I'm creating an issue ticket to fix the test in this project after completion of moving this to lts version. Can I do that in that ticket?

Comment (ashifkhn) [reply __tests__/ConnectionScreen-test.tsx:13]: Discussed on call. This test will be written in a different issue ticket.

**Pull request merged**